crates/rattler_index/src/lib.rs

```diff
@@ -97,117 +127,317 @@ pub fn package_record_from_conda(file: &Path) -> std::io::Result<PackageRecord>
     ))
 }
 
-/// Create a new `repodata.json` for all packages in the given output folder. If
-/// `target_platform` is `Some`, only that specific subdir is indexed. Otherwise
-/// indexes all subdirs and creates a `repodata.json` for each.
-pub fn index(output_folder: &Path, target_platform: Option<&Platform>) -> std::io::Result<()> {
-    let entries = WalkDir::new(output_folder).into_iter();
-    let entries: Vec<(PathBuf, ArchiveType)> = entries
-        .filter_entry(|e| e.depth() <= 2)
-        .filter_map(Result::ok)
-        .filter_map(|e| {
-            ArchiveType::split_str(e.path().to_string_lossy().as_ref())
-                .map(|(p, t)| (PathBuf::from(format!("{}{}", p, t.extension())), t))
+async fn index_subdir(
+    subdir: Platform,
+    op: Operator,
+    force: bool,
+    progress: MultiProgress,
+    semaphore: Arc<Semaphore>,
+) -> Result<()> {
+    let mut registered_packages: FxHashMap<String, PackageRecord> = HashMap::default();
+    if !force {
+        let repodata_path = format!("{subdir}/repodata.json");
+        let repodata_bytes = op.read(&repodata_path).await;
+        let repodata: RepoData = match repodata_bytes {
+            Ok(bytes) => serde_json::from_slice(&bytes.to_vec())?,
+            Err(e) => {
+                if e.kind() != opendal::ErrorKind::NotFound {
+                    return Err(e.into());
+                }
+                tracing::info!("Could not find repodata.json. Creating new one.");
+                RepoData {
+                    info: Some(ChannelInfo {
+                        subdir: subdir.to_string(),
+                        base_url: None,
```

Comment:
we should be able to infer the base url from the cli args, right?

Reply:
would that be `s3://my-bucket/my-channel` then? This field is never set in the repodata.json files that I have access to.

Reply:
We decided to omit this, similar to what conda-forge and artifactory do.

Reply:
this field actually does something else, you can point to a remote location for packages. for our use case, we can keep it `None`